src/main/java/net/dries007/tfc/common/entities/EntityHelpers.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.entities;
+
+import java.util.List;
+import java.util.Optional;
+
+import net.minecraft.advancements.CriteriaTriggers;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.syncher.EntityDataSerializer;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.ai.goal.*;
+import net.minecraft.world.entity.animal.Animal;
+import net.minecraft.world.entity.animal.Bucketable;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.*;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.level.Level;
+
+import net.dries007.tfc.common.entities.land.TFCAnimal;
+import net.dries007.tfc.common.entities.land.TFCAnimalProperties;
+import net.dries007.tfc.common.entities.ai.TFCAvoidEntityGoal;
+import net.dries007.tfc.util.calendar.Calendars;
+
+public class EntityHelpers
+{
+    public static final EntityDataSerializer<Long> LONG_ENTITY_SERIALIZER = new EntityDataSerializer<>()
+    {
+        @Override
+        public void write(FriendlyByteBuf pBuffer, Long pValue)
+        {
+            pBuffer.writeVarLong(pValue);
+        }
+        @Override
+        public Long read(FriendlyByteBuf pBuffer)
+        {
+            return pBuffer.readVarLong();
+        }
+        @Override
+        public Long copy(Long pValue)
+        {
+            return pValue;
+        }
+    };
+
+    public static void insertTFCAvoidGoal(PathfinderMob mob, GoalSelector selector, int priority)
+    {
+        selector.getAvailableGoals().removeIf(wrapped -> wrapped.getGoal() instanceof AvoidEntityGoal);
+        selector.addGoal(priority, new TFCAvoidEntityGoal<>(mob, Player.class, 8.0F, 5.0D, 5.4D));
+    }
+
+    public static void insertCommonPreyAI(TFCAnimal animal, GoalSelector goalSelector)
```

Comment:
This is also inconsistent since the previous method uses the `doSomethingGoal` but this is `doSomethingAI` when they look to be doing very similar tasks.